### PR TITLE
Implement single file support for map and routing

### PR DESCRIPTION
**Pull request by syershov**
@ygorshenin @bykoianko PTAL

### CODE REVIEW

**Comment by syershov** [comment]
DUDE
JTTP

File: generator/routing_generator.cpp [hunk 1]
```diff
@@ -234,7 +234,7 @@ void BuildCrossRoutingIndex(string const & baseDir, string const & countryName,
   routing::CrossRoutingContextWriter crossContext;
   FindCrossNodes(nodeData, osm2ft, countries, countryName, crossContext);
 
-  string const mwmRoutingPath = localFile.GetPath(MapOptions::CarRouting);
+  string const mwmRoutingPath = localFile.GetPath(MapOptions::CarRouting) + ROUTING_FILE_EXTENSION;
```
Comment (ygorshenin) [inline generator/routing_generator.cpp]: Why? GetPath(MapOptions::CarRouting) will return you a path with ROUTING_FILE_EXTENSION at the end.

File: generator/routing_generator.cpp [hunk 2]
```diff
@@ -413,19 +413,23 @@ void BuildRoutingIndex(string const & baseDir, string const & countryName, strin
...
-    // Write version for routing file that is equal to correspondent mwm file.
-    FilesContainerR mwmCont(localFile.GetPath(MapOptions::Map));
+  CHECK(my::CopyFileX(mwmPath, fPath), ("Can't copy", mwmPath, "to", fPath));
 
-    FileWriter w = routingCont.GetWriter(VERSION_FILE_TAG);
-    ReaderSource<ModelReaderPtr> src(mwmCont.GetReader(VERSION_FILE_TAG));
-    rw::ReadAndWrite(src, w);
-    w.WritePaddingByEnd(4);
-  }
+  FilesContainerW routingCont(fPath , FileWriter::OP_WRITE_EXISTING);
```
Comment (ygorshenin) [inline generator/routing_generator.cpp]: clang-format please

File: generator/routing_generator.cpp [hunk 3]
```diff
@@ -413,19 +413,23 @@ void BuildRoutingIndex(string const & baseDir, string const & countryName, strin
               "Multiple:", multiple, "Equal:", equal));
 
   LOG(LINFO, ("Collect all data into one file..."));
-  string const fPath = localFile.GetPath(MapOptions::CarRouting);
 
-  FilesContainerW routingCont(fPath /*, FileWriter::OP_APPEND*/);
+  string const mwmPath = localFile.GetPath(MapOptions::Map);
+  string const fPath = mwmPath + ROUTING_FILE_EXTENSION;
```
Comment (ygorshenin) [inline generator/routing_generator.cpp]: What does fPath mean? Could you please give it a more readable name? Why not routingPath?

Comment (gardster) [inline generator/routing_generator.cpp]: or localFile.GetPath(MapOptions::CarRouting) at least.

File: generator/routing_generator.cpp [hunk 4]
```diff
@@ -413,19 +413,23 @@ void BuildRoutingIndex(string const & baseDir, string const & countryName, strin
...
+  CHECK(my::CopyFileX(mwmPath, fPath), ("Can't copy", mwmPath, "to", fPath));
 
-    FileWriter w = routingCont.GetWriter(VERSION_FILE_TAG);
-    ReaderSource<ModelReaderPtr> src(mwmCont.GetReader(VERSION_FILE_TAG));
-    rw::ReadAndWrite(src, w);
-    w.WritePaddingByEnd(4);
-  }
+  FilesContainerW routingCont(fPath , FileWriter::OP_WRITE_EXISTING);
+
+//  {
```
Comment (gardster) [inline generator/routing_generator.cpp]: May be it is time to remove this useless code?

File: platform/local_country_file.cpp [hunk 5]
```diff
@@ -57,7 +57,9 @@ void LocalCountryFile::DeleteFromDisk(MapOptions files) const
 
 string LocalCountryFile::GetPath(MapOptions file) const
 {
-  return my::JoinFoldersToPath(m_directory, m_countryFile.GetNameWithExt(file));
+  // TODO (mpimenov): Refactor with MwmTraits after merge new-search branch.
+  bool singleFile = GetVersion() > 151126;
```
Comment (ygorshenin) [inline platform/local_country_file.cpp]: I'd prefer to leave CountryFile::GetPath() as is and to move the logic here:
```c++
if (file == MapOptions::Map || file == MapOptions::CarRouting)
{
  if (singleFile)
    return my::JoinFoldersToPath(m_directory, m_countryFile.GetNameWithExt(MapOptions::Map));
  return my::JoinFoldersToPath(m_directory, m_countryFile.GetNameWithExt(file);
}
```

Comment (syershov) [inline platform/local_country_file.cpp]: done

**Comment by syershov** [comment]
PTAL

**Comment by ygorshenin** [comment]
LGTM

**Comment by gardster** [comment]
LGTM

**Comment by bykoianko** [comment]
Давай сольем два последние коммита. 
И LGTM.

**Comment by syershov** [comment]
DUDE
JTTP

**Comment by syershov** [comment]
@bykoianko PTAL & merge

**Pull request merged**